File: SmallestDigitInNum_RecursiveApproach/Min.ts
/*
    Author : Pakshal Shashikant Jain 
    Date : 14/5/2024
    Program : Write a recursive program which accept number from user and return
              smallest digit

              Input : 87983
              Output : 3  
*/

//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

class MinDigit
{
    //Characteristics of Class
    imin : number;
    ino : number;

    //Constructor of Class
    constructor(value : number)
    {
        this.ino = value;
        this.imin = this.ino % 10;
        this.ino = Math.floor(this.ino / 10);
    }

    //Behaviour of Class To Find Smallest Digit in Entered Number
    Min() : number 
    {
        var Digit : number = 0;
        
        //Logic To Find Smallest Digit
        Digit = this.ino % 10;
        if(Digit < this.imin)
        {
            this.imin = Digit;
        }
        this.ino = Math.floor(this.ino / 10);

        if(this.ino != 0)
        {
            //Recursive Call To Behaviour of Class
            this.Min();
        }

        return this.imin;
    }
}

//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Main Function
function main() 
{
    var ino : number = 82987;
    var iret : number = 0;
    
    console.log("Jay Ganesh.......");

    //Creating object of MinDigit Class
    var mobj : MinDigit = new MinDigit(ino);
    //Call To Behaviour of Class
    iret = mobj.Min();

    //Printing Smallest Digit
    console.log("Smallest Digit in Entered Number %d is : %d",ino,iret);
}

////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
                    
//Call To Main Function
main();

File: SmallestDigitInNum_RecursiveApproach/Min.py
"""
    Author : Pakshal Shashikant Jain 
    Date : 14/5/2024
    Program : Write a recursive program which accept number from user and return
              smallest digit

              Input : 87983
              Output : 3  
"""

####################################################################################################################################

class MinDigit :
    #Characteristics of Class
    imin = 0;
    ino = 0;
    
    #Constructor of Class
    def __init__(self,ivalue) :
        self.ino = ivalue;
        self.imin = self.ino % 10;
        self.ino = self.ino // 10;

    #Behaviour of Class To Find Smallest Digit
    def Min(self) :
        Digit = 0;

        #Logic to Find
        Digit = self.ino % 10;
        if(Digit < self.imin) :
            self.imin = Digit;
        self.ino = self.ino // 10;

        if(self.ino != 0) :
            self.Min();
        
        return self.imin;

################################################################################################################################################

#Main Function
def main() :
    ino = 0;
    iret = 0;

    print("Jay Ganesh....");

    #Taking Input From User
    print("Enter One Number : ");
    ino = int(input());
    
    #Creating object of MinDigit Class
    mobj = MinDigit(ino);

    #Call To Behaviour of Class
    iret = mobj.Min();

    #Printing Smallest Digit
    print("Smallest Digit in Entered Number %d is : %d"%(ino,iret));

###################################################################################################################################################

#Call Back Method To Call Main Function
if __name__ == "__main__" :
    #Call To Main Function
    main();